project/MAINTAINERS.md

```diff
@@ -83,6 +83,99 @@ this (see below "Who decides what?")
     contributor on for weeks or months, having them make many changes to a PR
     that will eventually be rejected.
 
+## Pull request review workflow
+
+The main goal of having a PR review workflow that all maintainers follow
+is to have more consistency across maintainers, have better visibility in
+the pipeline of pull requests and thus increase the efficiency of pull request
+reviews.
+
+Think of this as a flexible workflow that uses common sense to
+bring some useful structure to the review process. If you nevertheless do
+experience problems inherent to this workflow, please discuss with
+maintainers on how to improve and open a pull request to this section.
+
+A pull request can be in 5 distinct states, for which there are corresponding
+labels that need to be applied. At any step of the way, a PR can be closed
+or return to an earlier state if needed (e.g. back to design review during code
+review).
+
+### Triage (*no label*)
+
+Maintainers are expected to triage new incoming pull requests by adding
+the correct labels (e.g. `1-design-review`) potentially skipping some steps
+depending on the kind of pull request. Use common sense for judging.
+
+Checking for DCO should be done at this stage.
+
+PRs with new user-facing features should follow all steps detailed below.
+However, performance optimizations or internal features (e.g. changes in
+/project) can skip Design review and/or Documentation review steps.
+
+Bugfixes can skip Design review if the bug is trivial. However, design review
+on bugfixes does not always need to be skipped: if a behavior is being
+mistaken for a bug. Rule of thumb: if you are not sure apply `1-design-review`
+label. If you reproduced the bug yourself, apply `2-code-review` directly.
+
+Proposals with only documentation changes, should get the `1-design-review`
+applied.
+
+Other documentation-only PRs should get the `3-docs-review` label applied
+directly.
+
+If an owner, responsible for closing or merging, can be assigned to the PR,
+the better.
+
+### Design review (`1-design-review`)
+
+Maintainers are expected to comment on the design of the pull request.
+Review of documentation is expected only in the context of design validation,
+not for stylistic changes.
+
+Ideally, documentation should reflect the expected behavior of the code.
+No code review should take place in this step.
+
+Once design is approved, a maintainer should remove this label and add
+the next appropriate label.
+
+In the case of Proposals with only documentation changes, the `3-docs-review`
+label should be applied.
+
+For changes that involve code, the `2-code-review` label should be applied.
+
+### Code review (`2-code-review`)
+
+Maintainers are expected to review the code and ensure that it is good
+quality and in accordance with the documentation.
+
+If documentation is absent but expected, maintainers should ask for documentation.
```

Comment:
If docs (from a design perspective) are done in step 1, then I would think missing docs should also be noticed in step 1, otherwise we'd need to move from step 2 back to step 1.